tests/python/network_spammer.py

```diff
@@ -0,0 +1,236 @@
+        builders_queue.put(b)
+
+    rounds = math.ceil(length // AVG_BLOCK_TIME)
+    for rnd in range(rounds):
+        round_builders = [builders_queue.get() for _ in range(tx_per_ledger)]
+
+        coroutines.append(spam_round(env, prioritizers,
+                                     round_builders[:TX_SET_SIZE - 1],
+                                     round_builders[TX_SET_SIZE - 1:tx_per_ledger],
+                                     rnd))
+
+        # push this round's builders to the back of the queue.
+        # they will be reused in the next LEDGER_BUFFER ledger
+        for b in round_builders:
+            builders_queue.put(b)
+
+    return asyncio.gather(*coroutines)
+
+
+# all transactions are payments to the same address
+#
+# TODO can this potentially create deadlocks when applying them?
+async def spam_round(env, prioritizers: List[Keypair], prioritized_builders, unprioritized_builders, rnd):
+    """Send prioritized and unprioritzied transactions for a single ledger.
+
+    All prioritized transactions are expected to be included in the next ledger.
+    Only one out of all unprioritized transactions is expected to be included in the next ledger.
+
+    Return a result list and hash of all submitted transactions.
+    """
+    payment_amount = 1
+    payment_dest = prioritizers[0]  # all transactions are payments to the same address
+
+    # this round should be started only after all previous ledgers have been
+    # added, so we sleep until then
+    await asyncio.sleep(rnd * AVG_BLOCK_TIME)
+
+    # generate unprioritized payment transactions
+    # we submit them first because we want to test if prioritized transactions
+    # actually get priority over them
+    tx_metadata = {}
+    xdrs = []
+    for i, _ in enumerate(unprioritized_builders):
+        builder = unprioritized_builders[i]
+        builder.append_payment_op(payment_dest.public_address, str(payment_amount))
+        builder.sign(builder.keypair.seed().decode())
+
+        xdrs.append(builder.gen_xdr())
+
+        # cache tx hash for later review
+        tx_metadata[builder.hash_hex()] = {'round': rnd,
+                                           'prioritized': False}
+
+    # generate prioritized transactions
+    for i, _ in enumerate(prioritized_builders):
+        builder = prioritized_builders[i]
+        builder.append_payment_op(payment_dest.public_address, str(payment_amount))
+        builder.sign(builder.keypair.seed().decode())
+
+        # prioritize transaction by adding a prioritizer signature
+        builder.sign(prioritizers[i].secret_seed)
+
+        xdrs.append(builder.gen_xdr())
+
+        tx_metadata[builder.hash_hex()] = {'round': rnd,
+                                           'prioritized': True}
+
+    # remember submission time and send txs
+    submission_time = time.time()
+
+    # some transactions can fail with HTTP 500 Server Error or HTTP 504 Server
+    # Timeout, so don't raise an exception if this happens
```

Comment:
why is this expected?

Reply:
the stress test submits more txs than the network can process, so a lot of txs will time out